lib/grid.rb

```diff
@@ -1,24 +1,38 @@
 class Grid
-  def template
-      <<~GRID
-          1  2  3
-         __ __ __
-      A |  |  |  |
-        |__|__|__|
-      B |  |  |  |
-        |__|__|__|
-      C |  |  |  |
-        |__|__|__|
-      GRID
-    end
+  attr_accessor :structure
 
-  def coordinate_valid?(coordinate)
-    coordinates.include?(coordinate)
+  def initialize
+    @structure = grid_structure
+  end
+
+  def mark(coordinate)
+    structure[coordinate] = 'X'
+  end
+  
+  def coordinate_invalid?(coordinate)
+    !structure.key?(coordinate)
+  end
+
+  def get_value(coordinate)
+    structure[coordinate]
+  end
+
+  def current_grid
+    <<~GRID
+       1  2  3
+       __ __ __
+    A |#{structure["A1"]} |#{structure["A2"]} |#{structure["A3"]} |
+      |__|__|__|
+    B |#{structure["B1"]} |#{structure["B2"]} |#{structure["B3"]} |
+      |__|__|__|
+    C |#{structure["C1"]} |#{structure["C2"]} |#{structure["C3"]} |
+      |__|__|__|
+    GRID
```

Comment:
I'm still not sure how to separate these things with how it's set up. Do you have suggestions?

Reply:
See comment above!